sympy/solvers/tests/test_ode.py

```diff
@@ -3063,6 +3064,25 @@ def test_sysode_linear_neq_order1():
     assert dsolve(eq, simplify=False) == sols_eq
 
 
+def test_order_reducing_substitution():
+    eqn = Eq(x*Derivative(f(x), x)**2 + Derivative(f(x), x, 2))
+    sol = Eq(f(x),
+             C1 - sqrt(-1/C1)*log(-C1*sqrt(-1/C1) + x) + sqrt(-1/C1)*log(C1*sqrt(-1/C1) + x))
+    assert checkodesol(eqn, sol, order=2, solve_for_func=False)[0]
+    assert sol == dsolve(eqn, f(x), hint='order_reducing_substitution')
+    assert sol == dsolve(eqn, f(x))
+
+    w, x, y= symbols('w x y')
```

Comment:
because you created x here, you lost access to the already defined x in the lines above (so they raise an error). So only define
```
w, y= symbols('w y')
```

Reply:
Also "reducible" is to be changed to.

Reply:
Well these are two different cases so the code that I had added has been already worked upon
till it gets to your code so that x even if replaced shouldn't do any bad.Isn't it?